docs/pipeline/triggers/OVERVIEW.md

```diff
@@ -0,0 +1,101 @@
+# Modyn Triggering
+
+Alongside the simple triggers, Modyn also provides complex triggers that can be used to trigger the training of a model.
+
+Despite the line being blurry, complex triggers are generally more sophisticated and require more information to be provided to the trigger - they most of the time cannot be pre-determined via a simple configuration entry but require
+reingested information from the ongoing pipeline run.
+
+Some complex triggers can only make decisions in batched intervals as they are cannot be efficiently computed
+on a per-sample basis. Here we can find the `DataDrift` and `Cost` based triggers.
```

Comment:
(I know this just got moved in this PR) Maybe we can add one sentence like: The input to the trigger is already batched by the supervisor, this batching is done for performance-reasons. The `BatchedTriggers` operate on a separate level of batching to decouple this (?).

IDK, actually I am not sure why we did not use the batching parameter from the supervisor in the first place, since both batchings are somewhat for performance reasons

Reply:
I'll add the comment that I think the two levels of batching make sense, at least in the experimental pipeline mode.
The supervisor-level batching is mostly motivated by the memory constraints of the host system. As we have quite a lot of storage available on the sgs-gpu machines, we can easily set these batches to a high number and avoid extra loops through the supervisor logic.
The batching on trigger levels is dependent on the policy type and motivated by both performance reasons (we cannot detect drift on every sample as detection is rather expensive), and semantic constraints (detecting drift on very small data batches or evaluating on a really small number of samples doesn't make sense)